core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -438,6 +444,7 @@ private[spark] class ExecutorAllocationManager(
     }
 
     override def onTaskStart(taskStart: SparkListenerTaskStart): Unit = {
+      numRunningTasks += 1
```

Comment:
Good point.  It doesn't need synchronization because only one thread is writing to it, but we should make it volatile.